File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> hash;
        int comp;
        for (int i = 0; i < nums.size(); i++) {
            comp = target - nums[i]; // 9-2 = 7
            if (hash.find(comp) != hash.end()) {
                return {hash[comp], i};
            }
            hash.insert({nums[i], i});
        }
        return {};
        
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        i = 0
        while i < len(nums):
            temp = target - nums[i]
            if temp in hash:
                return [i, hash[temp]]
            hash[nums[i]] = i
            i+=1
            